api/config/deployment/init_config.go

```diff
@@ -158,38 +159,73 @@ func (c InitConfig) Render() (string, error) {
 // AutomateConfig returns an AutomateConfig with the appropriate
 // fields from an InitConfig filled in
 func (c InitConfig) AutomateConfig() *AutomateConfig {
-	cfg := NewAutomateConfig()
-	cfg.Global.V1.Fqdn = w.String(c.Fqdn)
-	cfg.Global.V1.FrontendTls = []*global.FrontendTLSCredential{
+	// Fill out global config
+	g := global.DefaultGlobalConfig()
+	g.V1.Fqdn = w.String(c.Fqdn)
+	g.V1.FrontendTls = []*global.FrontendTLSCredential{
 		{
 			Cert: c.FrontendCert,
 			Key:  c.FrontendKey,
 		},
 	}
 
 	if c.ProxyHost != "" {
-		cfg.Global.V1.Proxy = &global.Proxy{}
-		cfg.Global.V1.Proxy.Host = w.String(c.ProxyHost)
+		g.V1.Proxy = &global.Proxy{}
+		g.V1.Proxy.Host = w.String(c.ProxyHost)
 		if c.ProxyPort != 0 {
-			cfg.Global.V1.Proxy.Port = w.Int32(c.ProxyPort)
+			g.V1.Proxy.Port = w.Int32(c.ProxyPort)
 		}
 		if c.ProxyUser != "" {
-			cfg.Global.V1.Proxy.User = w.String(c.ProxyUser)
+			g.V1.Proxy.User = w.String(c.ProxyUser)
 		}
 		if c.ProxyPassword != "" {
-			cfg.Global.V1.Proxy.Password = w.String(c.ProxyPassword)
+			g.V1.Proxy.Password = w.String(c.ProxyPassword)
 		}
 		if len(c.NoProxy) > 0 {
-			cfg.Global.V1.Proxy.NoProxy = c.NoProxy
+			g.V1.Proxy.NoProxy = c.NoProxy
 		}
 	}
 
-	cfg.Deployment.V1.Svc.Channel = w.String(c.Channel)
-	cfg.Deployment.V1.Svc.UpgradeStrategy = w.String(c.UpgradeStrategy)
-	cfg.Deployment.V1.Svc.DeploymentType = w.String(c.DeploymentType)
-	cfg.LicenseControl.V1.Svc.License = w.String(c.License)
-	cfg.Elasticsearch.V1.Sys.Runtime.Heapsize = w.String(c.ESHeapSize)
-	return cfg
+	// Fill out automate config
+	cfg := AutomateConfig{}
+	cfg.Global = g
+
+	cfg.SetChannel(c.Channel)                 // nolint: errcheck
```

Comment:
I have a mental block on what action I should be taking with the errors I am currently ignoring. (Originally we were creating a new AutomateConfig and setting, instead of overriding, exactly the same values in the struct.)